internal/exec/vendor_component_utils.go

```diff
@@ -22,6 +22,19 @@ import (
 	cp "github.com/otiai10/copy"
 )
 
+// findComponentConfigFile identifies the component vendoring config file (`component.yaml` or `component.yml`)
+func findComponentConfigFile(basePath, fileName string) (string, error) {
```

Comment:
Is there a reason we're not trying to solve this more across the board? I think not supporting `.yml` and `.yaml` extensions consistently is more confusing and error prone. 

We should support both extensions for any YAML file load.